imports.go

```diff
@@ -1,135 +1,167 @@
+	testImports = unique(testImports)
+	if recursive {
+		for _, i := range testImports {
+			testData, err := list(i)
 			if err != nil {
-				return err
+				return imports, nil
 			}
+			testDeps = append(testDeps, testData.Deps...)
 		}
+
+		deps = append(deps, testDeps...)
 	}
 
-	return nil
+	deps = append(deps, testImports...)
+	deps = unique(deps)
+
+	return deps, nil
 }
 
-func parseImports(recursive bool) ([]string, error) {
-	cwd, err := os.Getwd()
-	if err != nil {
-		return nil, err
-	}
+func filterPackages(packages []string) []string {
+	var imports []string
+	cwd, _ := os.Getwd()
 
-	var (
-		allImports = map[string]bool{}
-		imports    = []string{}
-	)
+	for _, importing := range packages {
+		importpath, err := getRootImportpath(importing)
+		if err != nil {
+			continue
+		}
 
-	filepath.Walk(
-		cwd, func(path string, info os.FileInfo, err error) error {
-			if err != nil {
-				return err
-			}
+		if inTests {
+			importpath = strings.Replace(importpath, "__blankd__", "localhost:60001", -1)
+		}
 
-			if filepath.Base(path) == ".git" ||
-				filepath.Dir(path) == filepath.Join(cwd, "vendor") {
-				return filepath.SkipDir
-			}
+		if isOwnPackage(importpath, cwd) {
+			continue
+		}
 
-			if path == filepath.Join(cwd, "vendor") {
-				return nil
-			}
+		// Skip anything that is already vendored upstream
+		if strings.Contains(importpath, "/vendor/") {
+			continue
+		}
 
-			if !info.IsDir() {
-				return nil
+		found := false
+		for _, imported := range imports {
+			if importpath == imported {
+				found = true
+				break
 			}
+		}
 
-			err = recursiveParseImports(
-				allImports,
-				".",
-				path,
-			)
-			if _, ok := err.(*build.NoGoError); ok {
-				return nil
-			}
+		if found {
+			continue
+		}
 
-			if err != nil {
-				log.Println(err)
-			}
+		imports = append(imports, importpath)
+	}
 
-			return nil
-		},
-	)
+	return imports
+}
 
-	for importing, standard := range allImports {
-		if !standard {
-			importpath, err := getRootImportpath(importing)
-			if err != nil {
-				continue
-			}
+func ensureDependenciesExist(packages []string, includeTestingDependencies bool) {
+	args := []string{"get"}
+	if includeTestingDependencies {
+		args = append(args, "-t")
+	}
+	args = append(args, packages...)
 
-			if inTests {
-				importpath = strings.Replace(importpath, "__blankd__", "localhost:60001", -1)
-			}
+	out, err := execute(exec.Command("go", args...))
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "%s\n", out)
+	}
+}
 
-			if isOwnPackage(importpath, cwd) {
-				continue
-			}
+func list(pkg string) (listOutput, error) {
+	data := listOutput{}
 
-			found := false
-			for _, imported := range imports {
-				if importpath == imported {
-					found = true
-					break
-				}
-			}
+	out, err := execute(exec.Command("go", "list", "-e", "-json", pkg))
+	if err != nil {
+		return data, err
+	}
 
-			if found {
-				continue
-			}
+	err = json.Unmarshal([]byte(out), &data)
+	if err != nil {
+		return data, err
```

Comment:
json.Unmarshal returns abmigious error when can't unmarshal json payload, so, please, add value from `out` variable to error message, like as following:
```
return data, fmt.Errof("can't unmarshal go list JSON output: %s\n%q", err, out)
```